uber/models.py

```diff
@@ -473,11 +473,20 @@ def jobs_for_signups(self):
             return [job.to_dict(fields) for job in jobs if job.restricted or frozenset(job.hours) not in restricted_hours]
 
         def guess_attendee_watchentry(self, attendee):
-            return self.query(WatchList).filter(and_(or_(WatchList.first_names.contains(attendee.first_name),
-                                                         and_(WatchList.email != '', WatchList.email == attendee.email),
-                                                         and_(WatchList.birthdate != None, WatchList.birthdate == attendee.birthdate)),
-                                                     WatchList.last_name == attendee.last_name,
-                                                     WatchList.active == True)).all()
+            return self.query(WatchList).filter(
+                    and_(
+                        or_(
+                            WatchList.first_names.contains(attendee.first_name),
+                            and_(WatchList.email != '', WatchList.email == attendee.email),
+                            *[WatchList.birthdate == attendee.birthdate] if
+                                WatchList.birthdate and WatchList.birthdate != '' and
```

Comment:
If that's what we mean to do then we need to move that condition out of the ``if`` block and into the ``*[]`` block.